### PR TITLE
Add justification for Environments section of checklist

### PR DESCRIPTION
**Pull request by yusufsheiqh**

### CODE REVIEW

**Comment by C-gyorfi** [comment]
[P15n - Context & justification - 3. Environments](https://trello.com/c/8wHd9ABB/62-p15n-context-justification-3-environments)

File: context-and-compliance/03-environments.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# 3 - Environments
+
+## 3.1 Application has corresponding environments and hosting
+
+### 3.1.1 `testing` environment
+
+#### Context & justification
+
+The `testing` environment, as the name suggests, is an environment for testing. This is where all development/testing is done. Hence, it should only be accessible to engineers and for internal delivery teams. Self-explanatory.
```
Comment (cybojenix) [inline context-and-compliance/03-environments.md]: Might be worth specifying "manual testing for the development team".

I've seen cases where development environments end up using resources from a hosted test environment. It's not pretty.

Comment (cybojenix) [inline context-and-compliance/03-environments.md]: When adding justification, we should assume the person doesn't know what the criteria is for. Removing the "Self explanatory" part.

```suggestion
The `testing` environment, as the name suggests, is an environment for testing. This is where all development/testing is done. Hence, it should only be accessible to engineers and for internal delivery teams.
```

Comment (cybojenix) [inline context-and-compliance/03-environments.md]: >Hence, it should only be accessible to engineers and for internal delivery teams

This implies setting up some form of authorisation barrier.
Suggest rewording to "should only be accessed by those within the delivery teams"

File: context-and-compliance/03-environments.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+#### Compliance
+
+* [ ] `testing` for internal delivery team; this environment should only be accessible to engineers
```
Comment (cybojenix) [inline context-and-compliance/03-environments.md]: ditto on wording; "should only be accessed by those within the delivery teams"

File: context-and-compliance/03-environments.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+### 3.1.2 `staging` environment
+
+#### Context & justification
+
+The `staging` environment is an environment typically used for external (customer, users, etc) reviews. The staging environment may also require showcasing to customers depending on feedback from reviews, however it is also on a feature by feature basis. This environment is additionally used for high level end-to-end (E2E) testing, as part of **Acceptance Test Driven Development (ATDD)**, which tests from the input of the system to the output of the system.
```
Comment (cybojenix) [inline context-and-compliance/03-environments.md]: >The staging environment may also require showcasing to customers depending on feedback from reviews, however it is also on a feature by feature basis

No need to have the "it is also" part

>however this should be on a feature-by-feature basis

File: context-and-compliance/03-environments.md (hunk 1 above)
Comment (yusufsheiqh) [reply context-and-compliance/03-environments.md]: Resolved in https://github.com/madetech/productionisation/pull/58/commits/8fd653e94c4c8d997663f95bdf49801fea5f08c8.

Comment (yusufsheiqh) [reply context-and-compliance/03-environments.md]: Resolved in https://github.com/madetech/productionisation/pull/58/commits/8fd653e94c4c8d997663f95bdf49801fea5f08c8.

File: context-and-compliance/03-environments.md (hunk 2 above)
Comment (yusufsheiqh) [reply context-and-compliance/03-environments.md]: Resolved in https://github.com/madetech/productionisation/pull/58/commits/376e6824e557e5f137c138b5cfadee9b5bba0e24.

File: context-and-compliance/03-environments.md (hunk 3 above)
Comment (yusufsheiqh) [reply context-and-compliance/03-environments.md]: Resolved in https://github.com/madetech/productionisation/pull/58/commits/109598ae69600f44cc6b160e1d22d55923d45045.
